packages/salesforcedx-vscode-apex/src/views/testOutline.ts

```diff
@@ -0,0 +1,666 @@
+  '..',
+  'resources',
+  'light',
+  'testFail.svg'
+);
+const LIGHT_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'light',
+  'testPass.svg'
+);
+
+const DARK_BLUE_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testNotRun.svg'
+);
+const DARK_RED_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testFail.svg'
+);
+const DARK_GREEN_BUTTON = ospath.join(
+  __filename,
+  '..',
+  '..',
+  '..',
+  '..',
+  'resources',
+  'dark',
+  'testPass.svg'
+);
+
+export type FullTestResult = {
+  summary: TestSummary;
+  tests: TestResult[];
+};
+
+export type TestSummary = {
+  outcome: string;
+  testsRan: number;
+  passing: number;
+  failing: number;
+  skipped: number;
+  passRate: string;
+  failRate: string;
+  testStartTime: string;
+  testExecutionTime: string;
+  testTotalTime: string;
+  commandTime: string;
+  hostname: string;
+  orgId: string;
+  username: string;
+  testRunId: string;
+  userId: string;
+};
+
+export type TestResult = {
+  ApexClass: ApexClass;
+  MethodName: string;
+  Outcome: string;
+  RunTime: number;
+  Message: string;
+  StackTrace: string;
+  FullName: string;
+};
+
+export type ApexClass = {
+  attributes: { type: string };
+  Id: string;
+  Name: string;
+  NamespacPrefix: string;
+};
+
+export type ApexGroupInfo = {
+  outcome: string;
+  summary: TestSummary;
+};
+
+export class ApexTestOutlineProvider implements vscode.TreeDataProvider<Test> {
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    Test | undefined
+  > = new vscode.EventEmitter<Test | undefined>();
+  public readonly onDidChangeTreeData: vscode.Event<Test | undefined> = this
+    ._onDidChangeTreeData.event;
+
+  private eventsEmitter = new events.EventEmitter();
+
+  private apexTestMap: Map<string, Test> = new Map<string, Test>();
+  private head: Test | null;
+  private static testStrings: Set<string> = new Set<string>();
+
+  constructor(
+    private path: string,
+    private apexClasses: vscode.Uri[],
+    private apexTestInfo: ApexTestRequestInfo[] | null
+  ) {
+    this.head = null;
+    this.getAllApexTests(this.path);
+    // Now, activate events that we need
+    this.eventsEmitter.on('Delete Folder', this.deleteFolderRecursive); // Activate event to delete folder
+    this.eventsEmitter.on('Show Highlight', this.updateSelection);
+  }
+
+  public getHead(): Test {
+    if (this.head === null) {
+      return this.getAllApexTests(this.path);
+    }
+    return this.head;
+  }
+
+  public getChildren(element: Test): Thenable<Test[]> {
+    if (element) {
+      return Promise.resolve(element.children);
+    } else {
+      if (this.head && this.head.children.length > 0) {
+        return Promise.resolve(this.head.children);
+      } else {
+        const emptyArray = new Array<ApexTest>();
+        emptyArray.push(new ApexTest('No tests in folder', null, 0));
```

Comment:
@jkalloor0701 - All messages should be externalized.
@ruthemmanuelle - Comments?

Reply:
<img width="296" alt="screen shot 2018-08-02 at 12 49 48 pm" src="https://user-images.githubusercontent.com/39964061/43604292-a2c3e4c8-9652-11e8-9bd5-e671b4d4b6f2.png">
The message is shown in the sidebar itself, and the user encounters it when there are no tests in the project

Reply:
Thanks, @jkalloor0701 . Would it make sense to give the user a bit more info here? We could add some language like this, which I adapted from a similar situation we encountered in the past.

No Tests Found
Your project doesn't contain any Apex test methods. To run Apex tests, open a project that contains methods with @isTest annotations or the testMethod keyword.

Reply:
I could add that. It might be a little long for that small amount of space. Can I have that be the description when you hover over?

Reply:
Sure, having the sentence be hover text works. In that case, please change the label to "No Apex Tests Found".